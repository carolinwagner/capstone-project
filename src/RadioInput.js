import React from 'react'
import styled from 'styled-components/macro'

const RadioInput = ({ question, register }) => {
  return (
    <div>
      {question.answerOptions.map((answerOption) => {
        return (
          <StyledRadioContainer key={answerOption.name}>
            <StyledRadioInput
              tabindex="0"
              name={question?.name || 'defaultRadioInput'}
              value={answerOption.name}
              id={answerOption.name}
              type="radio"
              ref={register(question.validationHookForm)}
              {...question?.validationNative}
            />
            <StyledLabel htmlFor={answerOption.name}>
              {answerOption.label}
            </StyledLabel>
          </StyledRadioContainer>
        )
      })}
    </div>
  )
}

export default RadioInput

const StyledRadioContainer = styled.div`
  margin-top: 20px;
  display: flex;
`

const StyledRadioInput = styled.input`
  align-self: start;
  margin: 10px 10px 0;
`
const StyledLabel = styled.label`
  line-height: 1.6;
`
