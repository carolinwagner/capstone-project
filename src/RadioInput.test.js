import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import RadioInput from './RadioInput'

const defaultQuestion = {
  answerOptions: [
    { label: 'a', name: 'a' },
    { label: 'b', name: 'b' },
  ],
}
describe('RadioInput', () => {
  it('renders RadioInput correctly', () => {
    const tree = renderer.create(<RadioInput question={defaultQuestion} />)
    expect(tree).toMatchSnapshot()
  })
})
