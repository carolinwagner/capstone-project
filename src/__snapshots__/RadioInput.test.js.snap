// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput renders RadioInput correctly 1`] = `
<div>
  <div>
    <input
      className="RadioInput__StyledRadioInput-zlk93r-0 jAuPYN"
      name="defaulRadioInput"
      type="radio"
      value="a"
    />
    <label
      htmlFor="a"
    >
      a
    </label>
  </div>
  <div>
    <input
      className="RadioInput__StyledRadioInput-zlk93r-0 jAuPYN"
      name="defaulRadioInput"
      type="radio"
      value="b"
    />
    <label
      htmlFor="b"
    >
      b
    </label>
  </div>
</div>
`;
